lua/lazyvim/plugins/extras/lang/kotlin.lua

```diff
@@ -0,0 +1,96 @@
+return {
+  -- Add linting, debug adapter, language server
+  {
+    "williamboman/mason.nvim",
+    opts = function(_, opts)
+      vim.list_extend(opts.ensure_installed or {}, {
+        "ktlint",
+        "kotlin-language-server",
```

Comment:
```suggestion
```
I believe this can be removed, as every server that is defined in `nvim-lspconfig.opts.servers` will be automatically installed.

Reply:
However it is "ensure_installed", so doesn't it help with verification? IMO it is better to verify 🙂

Reply:
It's just redundant, since it's already defined in `nvim-lspconfig`. Every server that is defined in `nvim-lspconfig` is automatically added to `mason-lspconfig`'s `ensure_installed` that takes care of the LSP servers' installation. For more info take a look [here](https://github.com/LazyVim/LazyVim/blob/97480dc5d2dbb717b45a351e0b04835f138a9094/lua/lazyvim/plugins/lsp/init.lua#L194-L216).

Reply:
If servers are defined within nvim-lspconfig, and they are automatically added to mason-lspconfig, then I think we can remove it.

However, everyone might need to have this concept (knowing that servers will be added to mason-lspconfig).

Let me refer to how other plugins of lang define language servers and see if we can implement it in a way that is as consistent as possible with other languages.

Reply:
> Every server that is defined in nvim-lspconfig is automatically added to mason-lspconfig's ensure_installed that takes care o

Thank you very much for the reminder! It really helped me notice this part!!!

Reply:
@Dogacel @dpetka2001 
Currently, I see that there are two approaches being used. 

Most of them are only defined in nvim-lspconfig, but for instance, markdown and helm are defined in mason + nvim-lspconfig.

From my perspective, since servers confirmed in nvim-lspconfig will be added to mason-lspconfig, I believe we can consider removing kotlin-language-server from ensured_installed.

What are your thoughts on this～～～?

Reply:
You can test it locally with your PR and see if the server will automatically install or not. Usually, `nvim-lspconfig` is where you define the LSP servers to be installed and `mason.nvim` the rest of the tools needed for a language (formatters and linters). 

Not saying this is absolutely how you have to do it. In my personal opinion, it's just redundant code, that's all. And I don't think it will matter in the end if you also leave it in Mason's `ensure_installed`. I just expressed my opinion. Nothing more than that and the one who will make the final decision is definitely the maintainer as I'm just a simple user like you.

Reply:
> Not saying this is absolutely how you have to do it. In my personal opinion, it's just redundant code, that's all. And I don't think it will matter in the end if you also leave it in Mason's ensure_installed. 
I agree, I'll remove the kotlin-language-server from the `ensure_installed` section in Mason.

I'll also test it in local to make sure.